token/cli/src/main.rs

```diff
@@ -1896,45 +1886,28 @@ async fn command_required_transfer_memos(
         existing_extensions.push(ExtensionType::MemoTransfer);
         let needed_account_len = ExtensionType::get_account_len::<Account>(&existing_extensions);
         if needed_account_len > current_account_len {
-            instructions.push(reallocate(
-                &config.program_id,
-                &token_account_address,
-                &config.fee_payer.pubkey(),
-                &owner,
-                &config.multisigner_pubkeys,
-                &existing_extensions,
-            )?);
+            token
+                .reallocate(
+                    &token_account_address,
+                    &owner,
+                    &[ExtensionType::MemoTransfer],
+                    &bulk_signers,
+                )
+                .await?;
```

Comment:
im undecided about this. theres a few approaches:
* reallocate as its own transaction, as i do here. this happens never more than once per account, so it feels fine
* always include a realloc instruction in the enable/disable memos methods in the token client. this is... tolerable... because process_reallocate bails early if its unnecessary, but i dont really like it
* fetch the account in the enable/disable memos methods in the token client. this is the most reasonable alternative, tho i dont strongly prefer it, because a) we cant do this in sign-only mode and the client shouldnt gate its modes, b) very few client methods perform fetches, and usually it will be unnecessary
* fuck up the interface, either by passing a realloc bool or adding a second method or something. this is dumb i dont like it

Reply:
It's a tough one... I think you chose the best approach